### PR TITLE
Discord Presence

### PR DESCRIPTION
**Pull request by OctopusProteins**
- Added support for a presence message as described in #90 and #142 .
- Added relevant settings for the presence cycler

Let me know if you need me to modify anything.

### CODE REVIEW

File: octoprint_discordremote/presence.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    def presence(self):
+        presence_cycle = {
+            0: "{}help".format(self.plugin.get_settings().get(["prefix"])),
+            1: "{}".format(self.generate_status())
+        }
+        while not self.discord.shutdown_event.is_set():                
+
+            if self.plugin.get_settings().get(['presence']):
+                presence_cycle[1] = "{}".format(self.generate_status())
+                self.discord.update_presence(presence_cycle[self.presence_cycle_id])
```
Comment (cameroncros) [inline octoprint_discordremote/presence.py:43]:
```suggestion
                self.discord.update_presence(presence_cycle[self.presence_cycle_id % len(presense_cycle)])
```

File: octoprint_discordremote/presence.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+                self.presence_cycle_id += 1
+                if self.presence_cycle_id == len(presence_cycle):
+                    self.presence_cycle_id = 0
+            else:
+                self.discord.update_presence("", disable=True)
+
+            for i in range(int(round(int(self.plugin.get_settings().get(['presence_cycle_time']))))):
```
Comment (cameroncros) [inline octoprint_discordremote/presence.py:51]: What is this int(round(int( stuff?
```suggestion
            for i in range(int(self.plugin.get_settings().get(['presence_cycle_time']))):
```

**Review by cameroncros — CHANGES_REQUESTED** [review]

File: octoprint_discordremote/presence.py (hunk 1 above)
Comment (OctopusProteins) [reply octoprint_discordremote/presence.py]: fixed by https://github.com/cameroncros/OctoPrint-DiscordRemote/pull/150/commits/a573705849abc9f4ca783c856f7126049ccd9cee

**Comment by OctopusProteins** [comment]
Made the changes you requested, Anything else?

File: octoprint_discordremote/presence.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    def presence(self):
+        presence_cycle = {
+            0: "{}help".format(self.plugin.get_settings().get(["prefix"])),
+            1: self.generate_status()
+        }
+        while not self.discord.shutdown_event.is_set():                
+
+            if self.plugin.get_settings().get(['presence']):
+                presence_cycle[1] = "{}".format(self.generate_status())
+                self.discord.update_presence(presence_cycle[self.presence_cycle_id] % len(presence_cycle))
```
Comment (cameroncros) [inline octoprint_discordremote/presence.py:43]:
```suggestion
                self.discord.update_presence(presence_cycle[self.presence_cycle_id % len(presence_cycle)])
```

**Review by cameroncros — APPROVED** [review]

**Pull request merged**